include/arch/arm/arch/32/mode/model/statedata.h

```diff
@@ -34,3 +34,6 @@ extern pde_t armUSGlobalPD[BIT(PD_INDEX_BITS)] VISIBLE;
 extern pte_t   armUSGlobalPT[BIT(PT_INDEX_BITS)]   VISIBLE;
 #endif /* CONFIG_ARM_HYPERVISOR_SUPPORT */
 
+#ifdef CONFIG_ARM_CORTEX_A8
+extern word_t tlbLockCount VISIBLE;
+#endif /* CONFIG_ARM_CORTEX_A8 */
```

Comment:
I'm tempted to define a `ARM_HAS_TLB_LOCK` instead, that currently is defined as `CONFIG_ARM_CORTEX_A8`. It is an accident that this is currently only implemented for Cortex A8, other CPUs do also have the feature, we just haven't used it there. If we ever want to change that, we don't want the `#ifdef`s to go out of sync.

Reply:
To me the whole TLB lock seems to be premature optimisation and should be removed altogether until there is proof that it gives system level improvement in performance, not only in seL4 microbenchmarks.

Reply:
I agree with small changes, i.e. I would not remove anything here. But the patch does introduce logic duplication where there wasn't any before. Unless there's an argument why I'm wrong about it being duplication, I think we should avoid that.